code/modules/logging/log_entry.dm

```diff
@@ -83,6 +83,25 @@ GENERAL_PROTECT_DATUM(/datum/log_entry)
 
 #define MANUAL_JSON_ENTRY(list, key, value) list.Add("\"[key]\":[(!isnull(value)) ? json_encode(value) : "null"]")
 
+#if DM_VERSION >= 515
+#define CHECK_AND_TRY_FILE_ERROR_RECOVERY(file) \
+	var/static/in_error_recovery = FALSE; \
+	if(!fexists(##file)) { \
+		if(in_error_recovery) { \
+			in_error_recovery = FALSE; \
+			CRASH("Failed to error recover log file: [file]"); \
+		}; \
+		in_error_recovery = TRUE; \
+		logger.Log(LOG_CATEGORY_INTERNAL_ERROR, "attempting to perform file error recovery: [file]"); \
+		logger.init_log_category(logger.log_categories[category]?.type); \
+		call(src, UNLINT(__PROC__))(arglist(args)); \
+		return; \
+	}; \
+	in_error_recovery = FALSE;
+#else
+#define CHECK_AND_TRY_FILE_ERROR_RECOVERY(file)
```

Comment:
I still think that despite us moving to 515 soon, we should not let logs to fuck itself up because we don't at the very least crash like we used to.